pkg/config/cni.go

```diff
@@ -18,29 +19,43 @@ const (
 	canalFlannelBackendType      = "canal_flannel_backend_type"
 	canalFlannelFlexVolPluginDir = "canal_flex_volume_plugin_dir"
 	canalFlannelInterface        = "canal_iface"
+
+	calicoFlexVolumePluginDir = "calico_flex_volume_plugin_dir"
 )
 
-type canalConfig struct {
+type CanalConfig struct {
 	Calico  map[string]string `json:"calico"`
 	Flannel map[string]string `json:"flannel"`
 }
 
-type calicoConfig struct {
+type CalicoConfig struct {
+	Installation CalicoInstallationSPec `json:"installation"`
+}
+
+type CalicoInstallationSPec struct {
```

Comment:
microscopic nit:
```suggestion
type CalicoInstallationSpec struct {
```

you would of course need to correct all references to such

Reply:
yup sure, fixing